src/edu/washington/escience/myria/io/UriSink.java

```diff
@@ -0,0 +1,31 @@
+package edu.washington.escience.myria.io;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.Objects;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class UriSink implements DataSink {
+
+  @JsonProperty
+  private final URI uri;
+
+  public UriSink(@JsonProperty(value = "uri", required = true) final String uri) {
+    this.uri = URI.create(Objects.requireNonNull(uri, "Parameter uri cannot be null"));
+  }
+
+  @Override
+  public OutputStream getOutputStream() throws IOException {
+    Configuration conf = new Configuration();
+
+    FileSystem fs = FileSystem.get(uri, conf);
```

Comment:
I don't think we want to have a separate HDFS `DataSink` impl, but maybe we could fail a bit earlier and more informatively for non-HDFS URIs? You could just compare `uri.getScheme()` to "hdfs" for now.